sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReader.java

```diff
@@ -87,9 +91,13 @@ public boolean hasNext() throws IOException, InterruptedException {
         readers.set(index, null);
         index++;
         currentReader = readers.get(index);
-        return currentReader.nextKeyValue();
+        boolean result = currentReader.nextKeyValue();
```

Comment:
why not use currentReader.hasNext()?

Reply:
And also, currentReader maybe null because in constructor it may be assigned to null

Reply:
currentReader hasn't hasNext() method.

Reply:
currentReader set null only when (0 == readers.size();

before this code, already add:

``` if (0 == readers.size()  {
      return false;
    }```